# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1234762705 18000
#      Mon Feb 16 00:38:25 2009 -0500
# Node ID d1eed318100f3ac3d370f3d7c7e11b2ce890539d
# Parent  86cfcf133a19501a772950d72841c6211a5abbb8
help.m: shorten default help text

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2009-02-16  John W. Eaton  <jwe@octave.org>
+
+	* help/help.m: Shorten default help text.
+
 2009-02-15  John W. Eaton  <jwe@octave.org>
 
 	* help/doc.m, help/help.m, help/lookfor.m, help/type.m,
 	help/which.m, image/colormap.m, miscellaneous/delete.m,
 	miscellaneous/dir.m, miscellaneous/edit.m,
 	miscellaneous/intwarning.m, miscellaneous/ls.m,
 	miscellaneous/mex.m, miscellaneous/mkoctfile.m,
 	miscellaneous/run.m, miscellaneous/what.m, path/savepath.m,
diff --git a/scripts/help/help.m b/scripts/help/help.m
--- a/scripts/help/help.m
+++ b/scripts/help/help.m
@@ -25,33 +25,37 @@
 ## 
 ## The help command can give you information about operators, but not the
 ## comma and semicolons that are used as command separators.  To get help
 ## for those, you must type @kbd{help comma} or @kbd{help semicolon}.
 ## @seealso{doc, which, lookfor}
 ## @end deftypefn
 
 function help (name)
+
   if (nargin == 0)
-    disp ("Help is available for the topics listed below.");
-    disp ("");
-    
-    disp ("*** operators:");
-    operators = __operators__ ();
-    disp (list_in_columns (operators (:, 1)));
 
-    disp ("*** reserved words:");
-    keywords = __keywords__ ();
-    disp (list_in_columns (keywords (:, 1)));
+    puts ("\n\
+  For help with individual commands and functions type\n\
+\n\
+    help NAME\n\
+\n\
+  (replace NAME with the name of the command or function you would\n\
+  like to learn more about).\n\
+\n\
+  For a more detailed introduction to GNU Octave, please consult the\n\
+  manual.  To read the manual from the prompt type\n\
+\n\
+    doc\n\
+\n\
+  GNU Octave is supported and developed by its user community.\n\
+  For more information visit http://www.octave.org.\n\n");
 
-    disp ("*** available functions:");
-    functions = __list_functions__ ();
-    disp (list_in_columns (functions));
-    
   elseif (nargin == 1 && ischar (name))
+
     ## Get help text
     [text, format] = get_help_text (name);
     
     ## Take action depending on help text format
     switch (lower (format))
       case "plain text"
         status = 0;
       case "texinfo"
@@ -66,26 +70,26 @@ function help (name)
           error ("help: `%s' not found\n", name);
         endif
       otherwise
         error ("help: internal error: unsupported help text format: '%s'\n", format);
     endswitch
     
     ## Print text
     if (status != 0)
-      warning ("makeinfo: Texinfo formatting filter exited abnormally");
-      warning ("makeinfo: raw Texinfo source of help text follows...\n");
+      warning ("makeinfo: Texinfo formatting filter exited abnormally; raw Texinfo source of help text follows...\n");
     endif
 
     which (name);
     printf ("\n%s\n%s", text, __additional_help_message__ ());
-    
+
   else
     error ("help: invalid input\n");
   endif
+
 endfunction
 
 function [text, status] = do_contents (name)
   text = "";
   status = 1;
 
   d = find_dir_in_path (name);
   if (!isempty (d))
@@ -105,9 +109,8 @@ function [text, status] = do_contents (n
           [text, status] = strip_html_tags (text);
       endswitch
     unwind_protect_cleanup
       ## Restore path
       path (p);
     end_unwind_protect
   endif
 endfunction
-
